utils/identify_hits.py

```diff
@@ -0,0 +1,80 @@
+from typing import Literal
+
+import polars as pl
+
+
+# creating a function that calculates the weighted sum
+def calculate_weighted_sum(scores_df: pl.DataFrame) -> pl.DataFrame:
+    """
+    Computes compound scores for each treatment based on the weighted sum of
+    both on_score and off_score, weighted by the ratio of the cluster (cluster
+    single-cells / total treatment single-cells).
+    """
+
+    # calculated weighted sum
+    compound_scores = (
+        scores_df.group_by("treatment")
+        .agg(
+            (
+                (pl.col("on_score") * pl.col("ratio")).sum()
+                + (pl.col("off_score") * pl.col("ratio")).sum()
+            ).alias("compound_score")
+        )
+        .sort("compound_score")
+    )
```

Comment:
Would it make sense to try a lazy interpretation of this?
```suggestion
        scores_df.lazy().group_by("treatment")
        .agg(
            (
                (pl.col("on_score") * pl.col("ratio")).sum()
                + (pl.col("off_score") * pl.col("ratio")).sum()
            ).alias("compound_score")
        )
        .sort("compound_score").collect()
    )
```